selene-vscode/src/extension.ts

```diff
@@ -160,9 +160,28 @@ export async function activate(context: vscode.ExtensionContext) {
         diagnosticsCollection.set(document.uri, diagnostics)
     }
 
-    vscode.workspace.onDidSaveTextDocument(lint)
+    let disposable: vscode.Disposable
+    function listenToChange() {
+        if (disposable) {
+            disposable.dispose()
+        }
+
+        const lintOn = vscode.workspace.getConfiguration("selene").get<string>("run")
```

Comment:
Not just string--make a real enum type.

Reply:
I'm not sure if this is desirable since I'd have to map each string out to the enum anyways. I could maybe create a function to map it out in `util.ts`?

Reply:
I'm not sure if this is desirable since I'd have to map each string out to the enum anyways. I could maybe create a function to map it out in util.ts?

Reply:
No, you wouldn't. 

```ts
enum BlaBlaBla {
    OnSave = "onSave",
    OnType = "onType",
}
```